src/test/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogComputerPublisherTest.java

```diff
@@ -0,0 +1,164 @@
+package org.datadog.jenkins.plugins.datadog.publishers;
+
+import java.util.Arrays;
+
+import org.datadog.jenkins.plugins.datadog.DatadogUtilities;
+import org.datadog.jenkins.plugins.datadog.clients.DatadogClientStub;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import hudson.EnvVars;
+import hudson.model.Messages;
+import hudson.model.Computer;
+import hudson.slaves.OfflineCause;
+
+public class DatadogComputerPublisherTest {
+    @Rule 
+    public JenkinsRule jenkins = new JenkinsRule();
+    
+    @Test
+    public void testJenkinsNodeMetrics() throws Exception {
+        DatadogClientStub client = new DatadogClientStub();
+        DatadogComputerPublisherTestWrapper computerPublisher = new DatadogComputerPublisherTestWrapper();
+        ((DatadogComputerPublisherTestWrapper)computerPublisher).setDatadogClient(client);
+        
+        String url = jenkins.getURL().toString();
+        String hostname = DatadogUtilities.getHostname(null);
+        String nodeHostname = null;
+        for (Computer computer: jenkins.jenkins.getComputers()){
+            nodeHostname = computer.getHostName();
+        }
+        
+        String[] expectedTags = new String[3];
+        expectedTags[0] = "node_name:master";
+        expectedTags[1] = "node_label:master";
+        expectedTags[2] = "jenkins_url:" + url;
+        
+        // The CI sets a hostname but we cannot set a hostname locally: https://javadoc.jenkins-ci.org/hudson/model/Computer.html#getHostName--
```

Comment:
This case was tricky- locally `computer.getHostname() `does not return anything, but on the CI it does. I couldn't find a good way to update this, see https://javadoc.jenkins-ci.org/hudson/model/Computer.html#getHostName--